mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -166,6 +167,7 @@ private[ml] object DecisionTreeClassificationModel {
         s" DecisionTreeClassificationModel (new API).  Algo is: ${oldModel.algo}")
     val rootNode = Node.fromOld(oldModel.topNode, categoricalFeatures)
     val uid = if (parent != null) parent.uid else Identifiable.randomUID("dtc")
-    new DecisionTreeClassificationModel(uid, rootNode, -1)
+    // Can't infer number of features from old model, so default to -1
+    new DecisionTreeClassificationModel(uid, rootNode, -1, -1)
```

Comment:
There isn't a way to infer the number of features from an old MLlib model since we only have access to the DT root node. I give it a dummy value of -1 here, but would appreciate feedback to see if there is a better way.